.github/workflows/ci-test.yml

```diff
@@ -29,3 +29,10 @@ jobs:
     - name: Build
       run: |
         make build
+    
+    - name: Spelling Check
+      uses: streetsidesoftware/cspell-action@v5.3.0
```

Comment:
I don't like using magic GH actions for linters. It should be possible to run the linter locally, e.g. `make spellcheck`

Reply:
Alright understood, I will make the changes and update again

Reply:
Wrote a python for spell check. Could you review?